### PR TITLE
Upgrade/rsb 634

### PR DESCRIPTION
**Pull request by simon998yang**
RSB-632
RSB-633
RSB-634
RSB-635
RSB-636
RSB-638
RSB-639

### CODE REVIEW

File: modules/flowable-ui-modeler/flowable-ui-modeler-app/oryx-editor-code/scripts/Core/StencilSet/stencilset.js [hunk 1]
```diff
@@ -399,6 +399,113 @@ ORYX.Core.StencilSet.StencilSet = Clazz.extend({
...
+          FLOWABLE.CONFIG.contextRoot + "/app/workflow/model/taskconfigs", {
+            asynchronous:false, method:'get',
+            onSuccess:function(result) {
+              customTaskConfigs=result.responseText.evalJSON();
+              },
+            onFailure:function(result) {}
+        });           
+    }
+    
+    // start added by simon to handle custom task
```
Comment (ruiboy) [inline modules/flowable-ui-modeler/flowable-ui-modeler-app/oryx-editor-code/scripts/Core/StencilSet/stencilset.js]: is it worth adding a custom string here to make such stuff easier to find eg
// rds - start added bysimon to handle custom task

File: modules/flowable-ui-modeler/flowable-ui-modeler-rest/src/main/java/org/flowable/app/rest/editor/ModelResource.java [hunk 2]
```diff
@@ -312,127 +315,23 @@ protected ModelRepresentation createNewModel(String name, String description, In
     return new ModelRepresentation(newModel);
   }
 
-  @RequestMapping(value = "/rest/model/exportForDeploy/{modelId}")
+  @RequestMapping(value = "/rest/models/{modelId}/exportForDeploy")
```
Comment (ruiboy) [inline modules/flowable-ui-modeler/flowable-ui-modeler-rest/src/main/java/org/flowable/app/rest/editor/ModelResource.java:306]: why would we change flowables existing request mappings?

File: modules/flowable-ui-modeler/flowable-ui-modeler-rest/src/main/java/org/flowable/app/rest/editor/ModelResource.java [hunk 3]
```diff
@@ -312,127 +315,23 @@ protected ModelRepresentation createNewModel(String name, String description, In
...
-              if (Model.MODEL_TYPE_FORM_RDS == childModel.getModelType()) {
-        formMap.put(childModel.getId(), childModel.getKey());
-
-              } else if (Model.MODEL_TYPE_DECISION_TABLE == childModel.getModelType()) {
-        decisionTableMap.put(childModel.getId(), childModel.getKey());
-      }
-    }
-          
-    BpmnModel model = new RDSBpmnJsonConverter().convertToBpmnModel(modelNode, formMap, decisionTableMap);
-    formsAsExtension(model);
```
Comment (ruiboy) [inline modules/flowable-ui-modeler/flowable-ui-modeler-rest/src/main/java/org/flowable/app/rest/editor/ModelResource.java:339]: Why have we got so much commented out code here?   Looks like flowable is putting the forms and decision tables in the XML.  Is this new to flowable? and conflicts with what we did to store forms in rds-namespaced elements in the past?

**Review by kevinlinxp — APPROVED** [review]

File: modules/flowable-ui-modeler/flowable-ui-modeler-rest/src/main/java/org/flowable/app/rest/editor/ModelResource.java (hunk 2 above)
Comment (simon998yang) [reply modules/flowable-ui-modeler/flowable-ui-modeler-rest/src/main/java/org/flowable/app/rest/editor/ModelResource.java:306]: This is not flowable existing mapping, I added this mapping, I changed it to make it more consistent like the existing flowable mapping

File: modules/flowable-ui-modeler/flowable-ui-modeler-app/oryx-editor-code/scripts/Core/StencilSet/stencilset.js (hunk 1 above)
Comment (simon998yang) [reply modules/flowable-ui-modeler/flowable-ui-modeler-app/oryx-editor-code/scripts/Core/StencilSet/stencilset.js]: fine, I will add rds as prefix to the comment.

File: modules/flowable-ui-modeler/flowable-ui-modeler-rest/src/main/java/org/flowable/app/rest/editor/ModelResource.java (hunk 3 above)
Comment (simon998yang) [reply modules/flowable-ui-modeler/flowable-ui-modeler-rest/src/main/java/org/flowable/app/rest/editor/ModelResource.java:339]: nope. I added this code in earlier PR, now it's been refactored into ModelServiceImpl.java

**Pull request merged**